backup_preload_mongo.sh

```diff
@@ -0,0 +1,1673 @@
+
+  status=$(oc get pvc cs-mongodump -n $FROM_NAMESPACE)
+  if [[ -z "$status" ]]; then
+    error "PVC cs-mongodump not found in $FROM_NAMESPACE"
+  fi
+
+  VOL=$(oc get pvc cs-mongodump -n $FROM_NAMESPACE  -o=jsonpath='{.spec.volumeName}')
+  if [[ -z "$VOL" ]]; then
+    error "Volume for pvc  cs-mongodump not found in $FROM_NAMESPACE"
+  fi
+
+  IMAGE=$(oc get pod icp-mongodb-0 -n $FROM_NAMESPACE  -o=jsonpath='{range .spec.containers[0]}{.image}{end}')
+  if [[ -z "$IMAGE" ]]; then
+    error "IMAGE for pod icp-mongodb-0 not found in $FROM_NAMESPACE"
+  fi
+
+  oc patch pv $VOL -p '{"spec": { "persistentVolumeReclaimPolicy" : "Retain" }}'
+  oc patch pv $VOL --type=merge -p '{"spec": {"claimRef":null}}'
+  oc patch pv $VOL --type json -p '[{ "op": "remove", "path": "/spec/claimRef" }]'
+  
+  oc delete pvc cs-mongodump -n $FROM_NAMESPACE --ignore-not-found --timeout=10s
+    if [ $? -ne 0 ]; then
+        info "Failed to delete pvc cs-mongodump, patching its finalizer to null..."
+        oc patch pvc cs-mongodump -n $FROM_NAMESPACE --type="json" -p '[{"op": "remove", "path":"/metadata/finalizers"}]'
+    fi
+
+  stgclass=$(oc get pvc mongodbdir-icp-mongodb-0 -n $FROM_NAMESPACE -o=jsonpath='{.spec.storageClassName}')
+  if [[ -z $stgclass ]]; then
+    error "Cannnot get storage class name from PVC mongodbdir-icp-mongodb-0 in $FROM_NAMESPACE"
+  fi
+
+  cat <<EOF >$TEMPFILE
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: cs-mongodump
+  namespace: $TO_NAMESPACE
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+  storageClassName: $stgclass
+  volumeMode: Filesystem
+  volumeName: $VOL
+EOF
+
+  oc create -f $TEMPFILE
+
+  status=$(oc get pvc cs-mongodump -n $TO_NAMESPACE --no-headers | awk '{print $2}')
+  while [[ "$status" != "Bound" ]]
+  do
+    namespace=$(oc get pv $VOL -o=jsonpath='{.spec.claimRef.namespace}')
+    if [[ $namespace != $TO_NAMESPACE ]]; then
+      oc patch pv $VOL --type=merge -p '{"spec": {"claimRef":null}}'
+    fi
+    info "Waiting for pvc cs-mongodump to bind"
+    sleep 10
+    status=$(oc get pvc cs-mongodump -n $TO_NAMESPACE --no-headers | awk '{print $2}')
+  done
+
+  success "Restored MongoDB volume moved to namespace $TO_NAMESPACE"
+} # swappvc
+
+
+#
+# Restore the mongodb in the to: namespace
+#
+function loadmongo() {
+  title "Restoring MongoDB to copy in namespace $TO_NAMESPACE"
+  msg "-----------------------------------------------------------------------"
+
+  currentns=$(oc project $TO_NAMESPACE -q)
+  if [[ "$currentns" -ne "$TO_NAMESPACE" ]]; then
+    error "Cannot switch to $TO_NAMESPACE"
+  fi
+
+  cat <<EOF >$TEMPFILE
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: mongodb-restore
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 20
+  template:
+    spec:
+      containers:
+      - name: icp-mongodb-restore
+        image: quay.io/opencloudio/ibm-mongodb:4.0.24
```

Comment:
It has to be the same image as in CP2

Reply:
same as backup job image, would there be somewhere to grab this from since it is a restore job?